src/main/kotlin/com/numberone/daepiro/domain/auth/controller/AuthController.kt

```diff
@@ -0,0 +1,48 @@
+package com.numberone.daepiro.domain.auth.controller
+
+import com.numberone.daepiro.domain.auth.api.AuthApi
+import com.numberone.daepiro.domain.auth.dto.request.AdminLoginRequest
+import com.numberone.daepiro.domain.auth.dto.request.RefreshTokenRequest
+import com.numberone.daepiro.domain.auth.dto.request.SocialLoginRequest
+import com.numberone.daepiro.domain.auth.dto.response.TokenResponse
+import com.numberone.daepiro.domain.auth.service.AuthService
+import com.numberone.daepiro.domain.user.enums.SocialPlatform
+import com.numberone.daepiro.global.dto.ApiResult
+import com.numberone.daepiro.global.exception.CustomErrorContext.UNSUPPORTED_PLATFORM
+import com.numberone.daepiro.global.exception.CustomException
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class AuthController(
+    private val authService: AuthService
+) : AuthApi {
+    override fun socialLogin(
+        platform: String,
+        request: SocialLoginRequest
+    ): ResponseEntity<ApiResult<TokenResponse>> {
```

Comment:
Service 에서는 TokenResponse 와 같은 dto 를 반환해서 Controller 로 올리고,

Controller 에서는 이를 ApiResult 로만 wrapping 하면 어떤 문제가 생기나요??

```suggestion
    ): ApiResult<TokenResponse> {
```

Reply:
api결과가 반환될때 HttpStatus 같은 body 외부값들을 커스터마이징하려는  목적으로 ResponseEntity를 사용했습니당